ivi/rohdeschwarz/rohdeschwarzBaseScope.py

```diff
@@ -606,15 +608,23 @@ def _set_channel_show_label(self, index, value):
     def _get_channel_enabled(self, index):
         index = ivi.get_index(self._channel_name, index)
         if not self._driver_operation_simulate and not self._get_cache_valid(index=index):
-            self._channel_enabled[index] = bool(int(self._ask("%s:state?" % self._channel_name[index])))
-            self._set_cache_valid(index=index)
+            if index < self._analog_channel_count:
+                self._channel_enabled[index] = bool(int(self._ask("%s:state?" % self._channel_name[index])))
+                self._set_cache_valid(index=index)
+            else:
+                self._channel_enabled[index] = bool(int(self._ask("Logic1:state?")))
```

Comment:
This only reads the first logic/digital channel, right? What happens if other digital (logic) channels are enabled?